### PR TITLE
refactor: migrate ext and verify-translations to typescript

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: src/ext.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (haxxmaxx) [inline src/ext.ts:2]: what does this mean?

File: src/ext.ts [hunk 2]
```diff
@@ -19,7 +23,7 @@ const columnExpressionItems = {
     translation: 'Object.Table.Columns.VisibilityCondition',
     defaultValue: { qv: '' },
     tid: 'visibilityCondition',
-    isExpression: (val) => typeof val === 'string' && val.trim().length > 0,
+    isExpression: (val: string) => typeof val === 'string' && val.trim().length > 0,
```
Comment (haxxmaxx) [inline src/ext.ts:26]: I'm pretty sure val can be something else since the function checks if it is a string

File: src/ext.ts (hunk 1 above)
Comment (LiKang6688) [reply src/ext.ts:2]: as qlik-modifiers does not contain types, it is a workaround to let typescript not complain.
I saw that people did the same for sn-table.
Screenshot 2022-06-29 at 22 24 57

File: src/ext.ts (hunk 2 above)
Comment (LiKang6688) [reply src/ext.ts:26]: right, then any idea what other types could be?

File: src/ext.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/ext.ts:2]: ok. If we notice that we need  // @ts-ignore in other places we can disable the `@typescript-eslint/ban-ts-comment` rule

File: src/ext.ts (hunk 2 above)
Comment (haxxmaxx) [reply src/ext.ts]: I guess yeah

**Comment by haxxmaxx** [comment]
So for a lot of the typings you are manually defining stuff that are in the api specs. I saw that you can generate types from the api specs so you don't have to do it twice, which might lead to mismatches. I think Thomas did it [here](https://github.com/qlik-oss/nebula.js/pull/752)

**Comment by LiKang6688** [comment]
> So for a lot of the typings you are manually defining stuff that are in the api specs. I saw that you can generate types from the api specs so you don't have to do it twice, which might lead to mismatches. I think Thomas did it [here](https://github.com/qlik-oss/nebula.js/pull/752)

Good point. I will look into the way to generate types from the api specs

**Pull request closed without merging**